src/Microsoft.DotNet.Interactive.HttpRequestParser/HttpRequestParser.cs

```diff
@@ -310,7 +459,38 @@ private HttpRequestNode ParseRequest()
                 {
                     return token;
                 }
-                else if (_currentTokenIndex + i == _tokens!.Count)
+
+                if (_currentTokenIndex + i < _tokens!.Count)
+                {
+                    i++;
+                    if(_currentTokenIndex + i >= _tokens.Count)
+                    {
+                        return null;
+                    }
+                    token = _tokens![_currentTokenIndex + i];
+                }
+                else
+                {
+                    break;
+                }
+            }
+
+            return null;
+        }
+
+        private HttpSyntaxToken? GetNextNodeType()
```

Comment:
This name looks inaccurate. How is this different from `GetNextSignificantToken`?